manifests/agent.pp

```diff
@@ -303,7 +302,7 @@
       $listen_ip = $::ipaddress
     }
   } else {
-    $listen_ip = $::ipaddress
+    notify { 'listenip is undef and will not be written to the configuration file' : }
```

Comment:
this is not a good idea, because this will show up in each run as an applied resource. We should probably leave the variable undefined or empty string, and check for that in the template.